frontend/src/pages/Wallet.js

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+// SDPX-FileCopyrightText: 2021 Que Le <b.le@tu-berlin.de>
+
+import {
+	Button,
+	Table,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableRow,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import appwriteApi from "../api/appwriteApi";
+import { useHistory } from "react-router-dom";
+import Grid from "@mui/material/Grid";
+import { useState } from "react";
+import detectEthereumProvider from "@metamask/detect-provider";
+import CenterFlexBoxMedium from "../components/CenterFlexBoxMedium";
+const { utils } = require( "ethers" );
+// import utils from "ethers";
+
+function AccountEntry({ data }) {
+	return 	<TableRow>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{data.pubAddress}
+		</TableCell>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{utils.formatEther(data.balance)}
+		</TableCell>
+	</TableRow>;
+}
+
+function EthAccountsDetail({ publicAddresses }) {
+	const [detailData, setDetailData] = useState([]);
+
+	async function fetchAccountsDetails(publicAddresses) {
+		const provider = await detectEthereumProvider();
+		if (!provider) {
```

Comment:
It would be nice if boolean values / predicates would be used instead of implicit type casts. In cases where the condition is not clear, an additional variable name can help.

Reply:
`!provider` is just the short hand notation for `provider === null`. I think this is fine and common practice.

Reply:
This case might be nearly trivial, but in general it also could mean `provider === undefined`, `provider === ""` etc. or even a combination of them (which would be even worse). I could just search for the documentation of the function, and it says, a promise can be resolved to `null`. I would likely be less uncomfortable with this if the implicit type casting rules of JavaScript wouldn't be broken so much and if it could only mean `provider === null`.

At least if we would like programming style to be consistent, `provider === null` is the better choice because writing conditions with implicit type casts becomes difficult in more complicated cases where the type is not clear.